heron/packing/src/java/com/twitter/heron/packing/binpacking/FirstFitDecreasingPacking.java

```diff
@@ -135,25 +145,46 @@ public void initialize(Config config, Config runtime) {
     this.maxContainerDisk = Long.parseLong(TopologyUtils.getConfigWithDefault(
         topologyConfig, com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_MAX_DISK_HINT,
         Long.toString(instanceDiskDefault * DEFAULT_NUMBER_INSTANCES_PER_CONTAINER)));
+
+    this.paddingPercentage = Integer.parseInt(TopologyUtils.getConfigWithDefault(
+        topologyConfig, com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_PADDING_PERCENTAGE,
+        Integer.toString(DEFAULT_CONTAINER_PADDING_PERCENTAGE)));
   }
 
+  /**
+   * Get a packing plan using First Fit Decreasing
+   *
+   * @return packing plan
+   */
   @Override
   public PackingPlan pack() {
     // Get the instances using FFD allocation
     Map<String, List<String>> ffdAllocation = getFFDAllocation();
-
-    if (ffdAllocation == null) {
-      return null;
-    }
     // Construct the PackingPlan
     Map<String, PackingPlan.ContainerPlan> containerPlanMap = new HashMap<>();
     Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
 
-    List<TopologyAPI.Config.KeyValue> topologyConfig = topology.getTopologyConfig().getKvsList();
-    int paddingPercentage = Integer.parseInt(TopologyUtils.getConfigWithDefault(
-        topologyConfig, com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_PADDING_PERCENTAGE,
-        Integer.toString(DEFAULT_CONTAINER_PADDING_PERCENTAGE)));
+    Resource resource = estimateResources(ffdAllocation, containerPlanMap, ramMap, false);
+
+    PackingPlan plan = new PackingPlan(topology.getId(), containerPlanMap, resource);
+    return plan;
+  }
+
+  @Override
+  public void close() {
+
+  }
 
+  /**
+   * Estimate the per instance and topology resources for the packing plan created using
+   * FFD placement
+   *
+   * @return Resources required
+   */
+  private Resource estimateResources(Map<String, List<String>> ffdAllocation,
```

Comment:
There are more changes in this class than required for a new initialize signature. Is that because you're refactoring to make things easier when implementing `IRepacking` or are these functional changes?